xarray/testing/strategies.py

```diff
@@ -0,0 +1,347 @@
+    max_size=5,
+)
+_attr_keys = _readable_strings
+_small_arrays = np_arrays(
+    shape=npst.array_shapes(
+        max_side=2,
+        max_dims=2,
+    )
+)
+_attr_values = st.none() | st.booleans() | _readable_strings | _small_arrays
+
+
+def attrs() -> st.SearchStrategy[Mapping[Hashable, Any]]:
+    """
+    Generates arbitrary valid attributes dictionaries for xarray objects.
+
+    The generated dictionaries can potentially be recursive.
+
+    Requires the hypothesis package to be installed.
+    """
+    return st.recursive(
+        st.dictionaries(_attr_keys, _attr_values),
+        lambda children: st.dictionaries(_attr_keys, children),
+        max_leaves=3,
+    )
+
+
+@st.composite
+def variables(
+    draw: st.DrawFn,
+    *,
+    array_strategy_fn: ArrayStrategyFn = None,
+    dims: st.SearchStrategy[Union[Sequence[Hashable], Mapping[Hashable, int]]] = None,
+    dtype: st.SearchStrategy[np.dtype] = numeric_dtypes(),
+    attrs: st.SearchStrategy[Mapping] = attrs(),
+) -> xr.Variable:
+    """
+    Generates arbitrary xarray.Variable objects.
+
+    Follows the basic signature of the xarray.Variable constructor, but allows passing alternative strategies to
+    generate either numpy-like array data or dimensions. Also allows specifying the shape or dtype of the wrapped array
+    up front.
+
+    Passing nothing will generate a completely arbitrary Variable (containing a numpy array).
+
+    Requires the hypothesis package to be installed.
+
+    Parameters
+    ----------
+    array_strategy_fn: Callable which returns a strategy generating array-likes, optional
+        Callable must accept shape and dtype kwargs if passed.
+        If array_strategy_fn is not passed then the shapes will be derived from the dims kwarg.
+        Default is to generate a numpy array of arbitrary shape, values and dtype.
+    dims: Strategy for generating the dimensions, optional
+        Can either be a strategy for generating a sequence of string dimension names,
+        or a strategy for generating a mapping of string dimension names to integer lengths along each dimension.
+        If provided as a mapping the array shape will be passed to array_strategy_fn.
+        Default is to generate arbitrary dimension names for each axis in data.
+    dtype: Strategy which generates np.dtype objects, optional
+        Will be passed in to array_strategy_fn.
+    attrs: Strategy which generates dicts, optional
+        Default is to generate a nested attributes dictionary containing arbitrary strings, booleans, integers, Nones,
+        and numpy arrays.
+
+    Returns
+    -------
+    variable_strategy
+        Strategy for generating xarray.Variable objects.
+
+    Raises
+    ------
+    TypeError
+        If custom strategies passed attempt to draw any examples which are not of the correct type.
+
+    Examples
+    --------
+    Generate completely arbitrary Variable objects backed by a numpy array:
+
+    >>> variables().example()
```

Comment:
When doctests runs on this is causes an error to be raised by hypothesis

```python
054     >>> variables().example()
UNEXPECTED EXCEPTION: NonInteractiveExampleWarning('The `.example()` method is good for exploring strategies, but should only be used interactively.  We recommend using `@given` for tests - it performs better, saves and replays failures to avoid flakiness, and reports minimal examples. (strategy: variables())')
```

How do I just write out the expected result in the docstring without actually running doctests?